packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts

```diff
@@ -0,0 +1,195 @@
+                    }
+                    break;
+                case EntityOperation.RemoveFromStart:
+                case EntityOperation.RemoveFromEnd:
+                case EntityOperation.Overwrite:
+                    const entity = event.entity;
+                    const id = entity.id;
+
+                    const delimiter = isDelimiter(entity);
+
+                    // If the current deleted entity is a delimiter, Remove the other delimiter pair and remove the Readonly entity between the delimiters.
+                    // Trigger the same event for the entity between the delimiters to allow other plugins to dispose resources accordingly.
+                    if (delimiter) {
+                        const [delimiterType, entity] = delimiter;
+
+                        if (delimiterType == DelimiterType.After) {
+                            const elBefore = editor.queryElements(getEntitySelector(id))[0];
+
+                            if (elBefore) {
+                                editor.select(
+                                    createRange(
+                                        new Position(elBefore, 0),
+                                        new Position(entity.wrapper, PositionType.After)
+                                    )
+                                );
+
+                                triggerEntityEvent(entity, event);
+                            }
+                        } else if (delimiterType == DelimiterType.Before) {
+                            const elAfter = editor.queryElements(getEntitySelector(id))[0];
+
+                            if (elAfter) {
+                                editor.select(
+                                    createRange(
+                                        new Position(entity.wrapper, PositionType.Before),
+                                        new Position(elAfter, 0)
+                                    )
+                                );
+                                triggerEntityEvent(entity, event, true /* deletedFromBefore */);
+                            }
+                        }
+                    }
+
+                    // If the entity removed is a readonly entity, try to remove delimiters around it.
+                    if (isReadOnly(entity)) {
+                        removeDelimiters(entity.wrapper);
+                    }
+                    break;
+
+                case EntityOperation.Click:
+                    const range = editor.getSelectionRange();
+                    // If the Entity is ReadOnly inline is clicked modify the selection to be the delimiter after. So we can see the cursor.
+                    if (isReadOnly(event.entity) && range?.collapsed) {
+                        const [_, delimiterAfter] = getDelimitersFromReadOnlyEntity(event.entity);
+
+                        if (delimiterAfter) {
+                            editor.select(createRange(delimiterAfter.wrapper, 1));
+                        }
```

Comment:
What if the readonly entity was designed to accept click event? In that case we should not change the selection.

Reply:
Good point, will remove this part